desktop/core/src/desktop/js/apps/storageBrowser/StorageDirectoryPage/StorageDirectoryActions/CreateAndUpload/CreateAndUploadAction.test.tsx

```diff
@@ -1,121 +1,145 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 import React from 'react';
-import { render, screen, fireEvent, waitFor, act } from '@testing-library/react';
+import { render, screen, fireEvent, waitFor, act, cleanup } from '@testing-library/react';
 import '@testing-library/jest-dom';
 import CreateAndUploadAction from './CreateAndUploadAction';
 import { CREATE_DIRECTORY_API_URL, CREATE_FILE_API_URL } from '../../../api';
+import * as storageUtils from '../../../../../utils/storageBrowserUtils';
 
 const mockSave = jest.fn();
 jest.mock('../../../../../utils/hooks/useSaveData/useSaveData', () => ({
   __esModule: true,
   default: jest.fn(() => ({
-    save: mockSave
+    save: mockSave,
+    loading: false,
+    error: undefined
   }))
 }));
 
-jest.mock('../../../../../utils/huePubSub', () => ({
-  __esModule: true,
-  publish: jest.fn()
-}));
-
 describe('CreateAndUploadAction', () => {
-  const currentPath = '/some/path';
+  const defaultPath = '/some/path';
   const onActionSuccess = jest.fn();
   const onActionError = jest.fn();
   const mockFilesUpload = jest.fn();
 
-  beforeEach(() => {
+  const setup = (path = defaultPath) =>
     render(
       <CreateAndUploadAction
-        currentPath={currentPath}
+        currentPath={path}
         onActionSuccess={onActionSuccess}
         onFilesUpload={mockFilesUpload}
         onActionError={onActionError}
       />
     );
+
+  const openDropdown = async () => {
+    await act(async () => fireEvent.click(screen.getByRole('button', { name: 'New' })));
+  };
```

Comment:
This works, but it feels unnecessary complex. Did you try to just use userEvent instead? Because then I think you can remove unnecessary act wrapping and simulate real user behavior better (keyboard/mouse).

```
const openDropdown = async (user) => {
  await user.click(screen.getByRole('button', { name: 'New' }));
};
```

You cam initialize user  (userEvent.setup()) in each test (or beforeEach).